src/main/java/com/iwaconsolti/league/demo/config/LeagueInitializer.java

```diff
@@ -0,0 +1,72 @@
+package com.iwaconsolti.league.demo.config;
+
+import com.iwaconsolti.league.demo.model.*;
+import com.iwaconsolti.league.demo.service.LeagueService;
+import com.iwaconsolti.league.demo.service.TeamService;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.Profile;
+import org.springframework.stereotype.Component;
+
+@Component
+@Profile("populated")
+public class LeagueInitializer {
+    private final LeagueService leagueService;
+    private final TeamService teamService;
+    @Value("${league.max-teams}")
+    private int maxTeams;
+
+    public LeagueInitializer(LeagueService leagueService, TeamService teamService) {
+        this.leagueService = leagueService;
+        this.teamService = teamService;
+        initializeData();
+    }
+
+    private void initializeData() {
+        League soccerLeague = new League();
+        soccerLeague.setName("Soccer League");
+        soccerLeague.setType(LeagueType.SOCCER);
+        soccerLeague.setMaxTeams(maxTeams);
```

Comment:
You are not initializing the maxTeams variable. This is because initializeData() is executed in the LeagueInitializer constructor, before Spring completes dependency injection on the class fields, including values ​​injected with @Value.

![image](https://github.com/user-attachments/assets/cd82faf0-dd95-4257-b880-773b3d678d88)

Reply:
Thanks! I use the  @PostConstruct and resolve the problem! (: